src/Microsoft.Data.Sqlite.Core/SqliteParameterBinder.cs

```diff
@@ -30,6 +42,29 @@ protected override void BindNull()
             => raw.sqlite3_bind_null(_stmt, _index);
 
         protected override void BindText(string value)
-            => raw.sqlite3_bind_text(_stmt, _index, value);
+        {
+            var text = value;
+            if (_size.HasValue && (value.Length > _size.Value))
+            {
+                text = value.Substring(0, _size.Value);
+            }
+
+            raw.sqlite3_bind_text(_stmt, _index, text);
+        }
+
+        internal static int GetSize(object value)
```

Comment:
I'd inline this logic with `SqliteParameter.Size` until something else needs it.